Tests/ApolloTests/StarWars/API.swift

```diff
@@ -54,59 +62,68 @@ public final class CreateReviewForEpisodeMutation: GraphQLMutation {
   public struct Data: GraphQLMappable {
     public let createReview: CreateReview?
 
-    public init(reader: GraphQLResultReader) throws {
-      createReview = try reader.optionalValue(for: Field(responseName: "createReview", arguments: ["episode": reader.variables["episode"], "review": reader.variables["review"]]))
+    public init(values: [Any?]) {
+      createReview = values[0] as! CreateReview?
     }
 
     public struct CreateReview: GraphQLMappable {
       public let __typename: String
       public let stars: Int
       public let commentary: String?
 
-      public init(reader: GraphQLResultReader) throws {
-        __typename = try reader.value(for: Field(responseName: "__typename"))
-        stars = try reader.value(for: Field(responseName: "stars"))
-        commentary = try reader.optionalValue(for: Field(responseName: "commentary"))
+      public init(values: [Any?]) {
+        __typename = values[0] as! String
+        stars = values[1] as! Int
+        commentary = values[2] as! String?
       }
     }
   }
 }
 
 public final class CreateAwesomeReviewMutation: GraphQLMutation {
-  public static let operationDefinition =
+  public static let operationString =
     "mutation CreateAwesomeReview {" +
     "  createReview(episode: JEDI, review: {stars: 10, commentary: \"This is awesome!\"}) {" +
     "    __typename" +
     "    stars" +
     "    commentary" +
     "  }" +
     "}"
+
+  public static let selectionSet: [Selection] = [
+    Field("createReview", arguments: ["episode": "JEDI", "review": ["stars": 10, "commentary": "This is awesome!"]], type: .object(Data.CreateReview.self), selectionSet: [
+      Field("__typename", type: .nonNull(.scalar(String.self))),
```

Comment:
How similar is this format to the JS GraphQL AST? Would it make sense to just make the format as close to the AST format as possible?